lib/config/config.js

```diff
@@ -283,6 +283,8 @@ function languageOptionsToJSON(languageOptions, objectKey = "languageOptions") {
 
 				if (name && hasMethod(value)) {
 					result[key] = name;
+				} else if (typeof value.toJSON === "function") {
```

Comment:
I think this doesn't cover the case when the language options object itself has a `.toJSON()` method (if we want to support that?).